packages/grid/_modules_/grid/hooks/features/sorting/useGridSorting.ts

```diff
@@ -150,22 +152,26 @@ export const useGridSorting = (
   );
 
   const buildComparatorList = React.useCallback(
-    (sortModel: GridSortModel): GridFieldComparatorList => {
-      const comparators = sortModel.map((item) => {
-        const column = apiRef.current.getColumn(item.field);
-        if (!column) {
-          throw new Error(`Error sorting: column with field '${item.field}' not found. `);
-        }
-        const comparator = isDesc(item.sort)
-          ? (
-              v1: GridCellValue,
-              v2: GridCellValue,
-              cellParams1: GridSortCellParams,
-              cellParams2: GridSortCellParams,
-            ) => -1 * column.sortComparator!(v1, v2, cellParams1, cellParams2)
-          : column.sortComparator!;
-        return { field: column.field, comparator };
-      });
+    (sortModel: GridSortModel): GridFieldComparator[] => {
+      const comparators = sortModel
+        .map((item) => {
+          const column = apiRef.current.getColumn(item.field);
+          if (!column) {
+            return null;
+          }
```

Comment:
I'm wondering if a warning could help developers to catch when they have a bug in their code. But this will be kind of spam for those who control the sort model. What do you think?

Reply:
Yes I think it would spam in valid use-cases.
But more globally, we have logger system that should answer those kind of issues and that is totally unusable because it spams way to much (I don't even add logging messages anymore). I think we could address this issue when cleaning the logger.